tripmining/model/traveler.py

```diff
@@ -6,41 +6,48 @@
 
 class Traveler:
 
-    def __init__(self, user: User, min_duration=7, min_density=0.2, filter_continent=''):
+    def __init__(self, user: User, checkin_streaks_gap=3, min_duration=7, min_density=0.2, filter_continent=''):
         self.countries = Countries()
         self.user_id = user.user_id
         self.dataset = user.dataset
         self.min_duration = min_duration
         self.min_density = min_density
         self.home_location: Location = user.home_location
-        self.ratio_tweets_home = (len(list(filter(lambda ci: user.checkin_at_home_location(ci), user.checkins))) / len(user.checkins))
+        self.ratio_tweets_home = (
+                len(list(filter(lambda ci: user.checkin_at_home_location(ci), user.checkins))) / len(user.checkins))
         self.home_country_code = user.home_country
         self.trips = set()
         self.filter_continent = filter_continent
-        self.__extract_trips(user)
+        self.__extract_trips(user, checkin_streaks_gap)
 
     def __get_filtered_sorted_checkins(self, user_checkins):
-        return filter(lambda x: self.filter_continent == '' or self.countries.get_by_iso(x.location.country_code).continent == self.filter_continent,
+        return filter(lambda x: self.filter_continent == '' or self.countries.get_by_iso(
+            x.location.country_code).continent == self.filter_continent,
                       user_checkins)
 
-    def __extract_trips(self, user: User):
+    def __extract_trips(self, user: User, checkin_streaks_gap):
         """
         This method segments the traveler's checkin stream into trips.
         :param user:
         :return: a list of trips
         """
+        last_home_checkin_before_trip = None
         current_travel_checkins = list()
         for checkin in self.__get_filtered_sorted_checkins(user.checkins):
+            if not last_home_checkin_before_trip:
+                last_home_checkin_before_trip = checkin
             if user.checkin_at_home_location(checkin) or user.is_last_checkin(checkin):
                 # Travel is ended due to home checkin or end of user checkin stream
                 # only append if there are checkins in the trip.
                 if current_travel_checkins:
-                    trip = Trip(self, current_travel_checkins)
+                    trip = Trip(self, current_travel_checkins, [last_home_checkin_before_trip, checkin],
```

Comment:
Please don't wrap these checkins as list. Make two different parameters out of it.

Reply:
last_checkin_before trip and first_checkin_after_trip are removed.

Fixed in the latest PR

Reply:
Hey @malinthasa, this was a misunderstanding. My idea was to keep these check-ins, but not as a list, but two separate parameters for `Trip`. This would look something like this then:
```
trip = Trip(self, current_travel_checkins, last_home_checkin_before_trip, checkin, checkin_streaks_gap)
```
Can you do so?